src/index.js

```diff
@@ -295,8 +298,18 @@ const snyker = async () => {
   console.log("[SNYKER: STARTING]");
 
   MAX_RETRIES = argv.retries || DEFAULT_RETRIES;
+  PRESERVE_INTEGRITY = argv["preserve-integrity"] || false;
+
+  // We need to determine whether we're using Yarn or NPM
+  // Prioritise "lockfile" flag, then check for yarn.lock, then package-lock.json
+  // If none of these files exist, default to yarn.lock
+  const lockFileName =
+    argv.lockfile || fs.existsSync(path.join(process.cwd(), "yarn.lock"))
+      ? "yarn.lock"
+      : fs.existsSync(path.join(process.cwd(), "package-lock.json"))
+        ? "package-lock.json"
+        : "yarn.lock";
```

Comment:
```suggestion
    argv.lockfile || ['package-lock.json', 'yarn.lock', 'npm-shrinkwrap.json', 'pnpm-lock.yaml'].find(file => fs.existsSync(path.join(process.cwd(), file));
```
...should do it

Reply:
This is a nicer approach for sure, so will make the change but caveat the above regarding the currently supported/tested lockfiles